File: starter-files/Python/part-01/task-03-format-output/blueprint_decoder.py
import re

def decode_blueprint(filename):
    with open(filename, "r") as file:
        content = file.read()

    pattern = r"\{\* (.*?) \*\}"
    secrets = re.findall(pattern, content)
    return secrets


# Enhanced version with error handling
# If file doesn't exist, return an empty list and print an error message
def decode_blueprint_safe(filename):
    try:
        return decode_blueprint(filename)
    except FileNotFoundError:
        print(f"Error: File '{filename}' not found.")
        return []


# Function to format and display secrets in a professional report
# Includes header, separator lines, numbered list, and footer
def display_secrets_report(secrets):
    # TODO: Create a separator string of '=' characters
    # TODO: Print header with title centered
    # TODO: Print total count of secrets
    # TODO: Loop through secrets and print each with numbering
    # TODO: Print footer separator
    pass


if __name__ == "__main__":
    secrets = decode_blueprint_safe("blueprint-data.txt")
    display_secrets_report(secrets)

File: starter-files/C#/part-01/task-03-format-output/BlueprintDecoder.cs
using System;
using System.IO;
using System.Text.RegularExpressions;
using System.Collections.Generic;

public class BlueprintDecoder
{
    public static List<string> DecodeBlueprint(string filename)
    {
        string content = File.ReadAllText(filename);

        string pattern = @"\{\* (.*?) \*\}";
        MatchCollection matches = Regex.Matches(content, pattern);

        List<string> secrets = new List<string>();
        foreach (Match match in matches)
        {
            secrets.Add(match.Groups[1].Value);
        }

        return secrets;
    }

    // Enhanced version with error handling
    // If file doesn't exist, return empty list and print error message
    public static List<string> DecodeBlueprintSafe(string filename)
    {
        try
        {
            return DecodeBlueprint(filename);
        }
        catch (FileNotFoundException)
        {
            Console.WriteLine($"Error: File '{filename}' not found.");
            return new List<string>();
        }
    }

    // Function to format and display secrets in a professional report
    // Includes header, separator lines, numbered list, and footer
    public static void DisplaySecretsReport(List<string> secrets)
    {
        // TODO: Create a separator string of '=' characters (50 chars wide)
        // TODO: Print header with title
        // TODO: Print total count of secrets
        // TODO: Loop through secrets and print each with numbering and alignment
        // TODO: Print footer separator
    }

    static void Main()
    {
        var secrets = DecodeBlueprintSafe("blueprint-data.txt");
        DisplaySecretsReport(secrets);
    }
}
